js/bitmart.js

```diff
@@ -500,141 +500,100 @@ module.exports = class bitmart extends Exchange {
-            // from the above API doc:
-            // quote_increment Minimum order price as well as the price increment
-            // price_min_precision Minimum price precision (digit) used to query price and kline
-            // price_max_precision Maximum price precision (digit) used to query price and kline
-            //
-            // the docs are wrong: https://github.com/ccxt/ccxt/issues/5612
-            //
-            const amountPrecision = this.safeNumber (contract, 'vol_unit');
-            const pricePrecision = this.safeNumber (contract, 'price_unit');
-            const contractType = this.safeValue (contract, 'contract_type');
+            const splitId = id.split (id, '_');
+            const splitIdEnd = this.safeString (splitId, 1);
+            let settle = 'USDT';
+            let symbol = base + '/' + quote + ':' + settle;
+            let type = 'swap';
+            let swap = true;
             let future = false;
-            let swap = false;
-            let type = 'contract';
-            let symbol = base + '/' + quote;
-            const expiry = this.parse8601 (this.safeString (contract, 'delive_at'));
-            if (contractType === 1) {
-                type = 'swap';
-                swap = true;
-                symbol = symbol + ':' + settle;
-            } else if (contractType === 2) {
-                type = 'future';
-                future = true;
-                symbol = symbol + ':' + settle + '-' + this.yymmdd (expiry, '');
+            if (splitIdEnd !== undefined) {
+                settle = 'BTC';
+                symbol = base + '/' + quote + ':' + settle;
+                if (splitIdEnd !== 'PERP') {
+                    const date = this.iso8601 (this.milliseconds ());
+                    const splitDate = this.split (date, '-');
+                    const year = splitDate[0];
+                    const shortYear = year.slice (0, 2);
+                    const expiry = shortYear + splitIdEnd;
+                    symbol = symbol + '-' + expiry;
+                    type = 'future';
+                    swap = false;
+                    future = true;
+                }
             }
-            const feeConfig = this.safeValue (market, 'fee_config', {});
             result.push ({
                 'id': id,
-                'numericId': numericId,
+                'numericId': undefined,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
                 'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
-                'settleId': settleId,
+                'settleId': undefined,
                 'type': type,
                 'spot': false,
                 'margin': false,
                 'swap': swap,
                 'future': future,
                 'option': false,
-                'active': undefined,
+                'active': true,
```

Comment:
I figure we can set this as true because we're fetching the tickers, and the ticker markets will all be active